src/NServiceBus.AcceptanceTests/Reliability/Outbox/When_a_duplicate_message_arrives.cs

```diff
@@ -0,0 +1,125 @@
+﻿namespace NServiceBus.AcceptanceTests.Reliability.Outbox
+{
+    using System;
+    using System.Threading.Tasks;
+    using NServiceBus.AcceptanceTesting;
+    using NServiceBus.AcceptanceTests.EndpointTemplates;
+    using NServiceBus.AcceptanceTests.ScenarioDescriptors;
+    using NServiceBus.Configuration.AdvanceExtensibility;
+    using NServiceBus.Pipeline;
+    using NUnit.Framework;
+
+    public class When_a_duplicate_message_arrives : NServiceBusAcceptanceTest
+    {
+        [Test]
+        public async Task It_does_not_dispatch_messages_aready_dispatched()
```

Comment:
Right now we lean on the persisters to clear the dispatched messages for us

https://github.com/Particular/NServiceBus.RavenDB/issues/81

We did have plans to make this explicit from a core perspective. @SzymonPobiega did we raise an issue for this?

In short: I think @SzymonPobiega is correct but the code is very confusing due to the implicitness